Products/CMFPlomino/PlominoForm.py

```diff
@@ -819,7 +819,12 @@ def applyHideWhen(self, doc=None, silent_error=True):
             hidewhenName = hidewhen.id
             try:
                 if doc is None:
-                    target = self
+                    db = self.getParentDatabase()
+                    target = getTemporaryDocument(
```

Comment:
We should do that outside the `for` loop instead of instantiate a new TempDoc for each hidewhen.